tests/components/garmin_connect/test_config_flow.py

```diff
@@ -0,0 +1,122 @@
+"""Test the Garmin Connect config flow."""
+from unittest.mock import patch
+
+from garminconnect import (
+    GarminConnectAuthenticationError,
+    GarminConnectConnectionError,
+    GarminConnectTooManyRequestsError,
+)
+import pytest
+
+from homeassistant import data_entry_flow
+from homeassistant.components.garmin_connect import config_flow
+from homeassistant.components.garmin_connect.const import DOMAIN
+from homeassistant.const import CONF_ID, CONF_PASSWORD, CONF_USERNAME
+
+from tests.common import MockConfigEntry
+
+MOCK_CONF = {
+    CONF_ID: "First Lastname",
+    CONF_USERNAME: "my@email.address",
+    CONF_PASSWORD: "mypassw0rd",
+}
+
+
+@pytest.fixture(name="mock_garmin_connect")
+def mock_garminconnect():
+    """Mock Garmin."""
+    with patch("homeassistant.components.garmin_connect.config_flow.Garmin") as garmin:
+        garmin().get_full_name.return_value = "First Lastname"
+        yield garmin
+
+
+def init_config_flow(hass):
+    """Init a configuration flow."""
+    flow = config_flow.GarminConnectConfigFlowHandler()
+    flow.hass = hass
+    return flow
+
+
+async def test_show_form(hass):
+    """Test that the form is served with no input."""
+    flow = init_config_flow(hass)
+
+    result = await flow.async_step_user(user_input=None)
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+
+
+async def test_step_user(hass, mock_garmin_connect):
+    """Test registering an integration and finishing flow works."""
+    flow = init_config_flow(hass)
+
+    result = await flow.async_step_user(user_input=MOCK_CONF)
```

Comment:
Start the flow with user input like this:
https://github.com/home-assistant/home-assistant/blob/80a55360dc0cba19f2b7ccfdc84ea369ae368f73/tests/components/hue/test_config_flow.py#L395-L397

The example above starts the import step. Replace that string as needed, with eg "user".

Reply:
This doesn't run, because:
`homeassistant.data_entry_flow.UnknownStep: Handler GarminConnectConfigFlowHandler doesn't support step import`
I didn't create it because I only have user input from integration config, no yaml based config, or is that wrong assumption?

Reply:
Wrote it like this:
```
    result = await hass.config_entries.flow.async_init(
        DOMAIN,
        context={"source": "user"},
        data={"username": "my@email.address", "password": "mypassw0rd"},
    )
```

Reply:
Yes, that's why I wrote that you replace the import string as needed with eg "user".

Reply:
You are right, overlooked it.